utils/bfd.mli

```diff
@@ -0,0 +1,22 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                        Nicolas Ojeda Bar, LexiFi                       *)
+(*                                                                        *)
+(*   Copyright 2020 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+type t
+
+val read: string -> t
```

Comment:
It would be nice if the failure modes where documented. One way to do it would be to expose a concrete datatype of possible errors (as an `error` variant), and then have a dedicated exception whose payload is `error`. (One could also return a `(t, error) result`, but this would be a much more invasive change.) Having an explicit data representation for errors makes it easier to inspect or reformat them if we want to later on.

Reply:
Agree, I will add an `error` type.

Reply:
Done.